README.md

```diff
@@ -1,55 +1,301 @@
+Hello other teams! This section is for you guys. Please reach out to us with problems or questions.
 
-1. Clones the googletest repository if it is not present
-2. Builds the libraries from the googletest source
+### General Interface Information
 
-### Build & Run Instructions
+Before describing the specific messages & formats we will support, here are some rules that apply to all messaging we will be involved in:
 
-#### Evaluation
+- **UDP Sockets**: All communication with our component will be through UDP socket communication. We will listen on one or more (TBD) sockets for messages, and will respond using UDP sockets. *If this is a significant problem for your component, discuss this with us, and we can try to work something else out*.
+> TCP will be slow, I don't think UDP will be a problem for our case, especially because we are all running within the same network.
+- **JSON**: All interactions will be done using JSON. This is for uniformity and ease of communication. There is one exception to this, the first 4 bytes will always be the length of the message (**NOT including this four byte length encoding**).
+- **Error Handling**: For the beta release, very minimal error handling will be implemented. If we receive a badly formatted message, we will ignore it. We will not respond with "success" or "failure" messages. This may result in the loss of some metrics or search data, however it will significantly reduce both ther server and developer load. If you send us a message and don't receive a response when expecting one, you will have to retry sending the message.
+> Not too sure if we want to not respond. Some data loss in our case is fine, and other teams probably wouldn't even be able to fix errors on the fly. Only problem here is if they are sending to the wrong socket or something. 
+>
+>Maybe we have separate `DEBUG` and `RELEASE` modes which respond? Or we could expand the idea of using `evaltool` as a proxy (see my note right above [Specific API Calls](#specific-api-calls)). The evaltool route is probably easier and better, that way we don't have to recompile when someone has a problem
 
-Enter the evaluation directory, and run make. This will create a binary titled `evaluation` in the `evaluation/bin` directory which can be executed:
+As mentioned above, all communication will be done in JSON (with the exception of the length encoding). Below there are several examples listed of what JSON format we will be accepting. Each of these must be wrapped in the following JSON:
 
 ```
-cd evaluatoin
-make
-./bin/evaluation
+{
+  "component": <Your Component Name>,
+  "function": <The API "function" name you are calling>,
+  "version": <The version of the API "function" that you are using>,
+  "message": <The message contents as a JSON object>
+}
 ```
 
-Alternatively, the `make run` shortcut will also do this
+Some more detail about each field:
```

Comment:
We may want different component identifiers for each subteam (i.e. webgraph UI vs mobile ui vs regular UI)